wallet/src/de/schildbach/wallet/ui/PaymentsFragment.kt

```diff
@@ -17,46 +17,50 @@
 package de.schildbach.wallet.ui
 
 import android.content.Context
-import android.content.Intent
 import android.os.Bundle
 import android.view.Menu
-import android.view.MenuItem
-import androidx.appcompat.widget.Toolbar
+import android.view.MenuInflater
+import android.view.View
+import androidx.fragment.app.Fragment
 import com.google.android.material.tabs.TabLayout
 import de.schildbach.wallet_test.R
 import kotlinx.android.synthetic.main.activity_payments.*
 
-class PaymentsActivity : GlobalFooterActivity() {
+class PaymentsFragment : Fragment(R.layout.activity_payments) {
 
     companion object {
         private const val PREFS_RECENT_TAB = "recent_tab"
-
-        private const val EXTRA_ACTIVE_TAB = "extra_active_tab"
+        private const val ARGS_ACTIVE_TAB = "extra_active_tab"
 
         const val ACTIVE_TAB_RECENT = -1
         const val ACTIVE_TAB_PAY = 0
         const val ACTIVE_TAB_RECEIVE = 1
 
         @JvmStatic
-        fun createIntent(context: Context, activeTab: Int): Intent {
-            val intent = Intent(context, PaymentsActivity::class.java)
-            intent.putExtra(EXTRA_ACTIVE_TAB, activeTab)
-            return intent
+        fun newInstance(activeTab: Int = ACTIVE_TAB_RECENT): PaymentsFragment {
+            val args = Bundle()
+            args.putInt(ARGS_ACTIVE_TAB, activeTab)
+
+            val instance = PaymentsFragment()
+            instance.arguments = args
+            return instance
         }
     }
 
     private var saveRecentTab = false
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentViewWithFooter(R.layout.activity_payments)
-        activateGotoButton()
+        setHasOptionsMenu(true)
+    }
 
-        val toolbar = findViewById<Toolbar>(R.id.toolbar)
-        setSupportActionBar(toolbar)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
 
-        setTitle(R.string.payments_title)
+        toolbar.setTitle(R.string.payments_title)
+        setupActionBarWithTitle(R.string.payments_title, false)
 
+        //TODO: Implement FragmentViewPager
```

Comment:
This is probably a future story?

Reply:
Yes, implementing a `FragmentViewPager` on this screen would allow for smoother switching between `Send` / `Received`.